zone/special_attacks.cpp

```diff
@@ -2153,104 +2166,133 @@ void Mob::Taunt(NPC *who, bool always_succeed, int chance_bonus, bool from_spell
+			taunt_chance += level_difference * 5;
 
-			taunt_chance /= 100.0f;
-			success = taunt_chance > zone->random.Real(0, 1);
+			if (taunt_chance > 50) {
+				taunt_chance = 50;
+			} else if (taunt_chance < 10) {
+				taunt_chance = 10;
+			}
+		}
+
+		// Taunt Chance Rule Bonus
+		taunt_chance += RuleI(Combat, TauntChanceBonus);
+	}
+
+	//success roll
+	success = zone->random.Roll(taunt_chance);
+
+	// Log result
+	LogHate(
+		"Taunter mob {} target npc {} taunt_chance [{}] success [{}] hate_top [{}]",
+		GetMobDescription(),
+		who->GetMobDescription(),
+		taunt_chance,
+		success ? "true" : "false",
+		hate_top ? hate_top->GetMobDescription() : "not found"
+	);
+
+	// Actual Taunting
+	if (success) {
+		if (hate_top && hate_top != this) {
+			int64 new_hate = (
+				(who->GetNPCHate(hate_top) - who->GetNPCHate(this)) +
+				bonus_hate +
+				RuleI(Combat, TauntOverAggro) +
+				1
+			);
 
 			LogHate(
-				"Taunter mob {} target npc {} taunt_chance [{}] success [{}] hate_top [{}]",
+				"Not Top Hate - Taunter [{}] Target [{}] Hated Top [{}] Hate Top Amt [{}] This Character Amt [{}] Bonus_Hate Amt [{}] TauntOverAggro Amt [{}] - Total [{}]",
 				GetMobDescription(),
 				who->GetMobDescription(),
-				taunt_chance,
-				success ? "true" : "false",
-				hate_top ? hate_top->GetMobDescription() : "not found"
+				hate_top->GetMobDescription(),
+				who->GetNPCHate(hate_top),
+				who->GetNPCHate(this),
+				bonus_hate,
+				RuleI(Combat, TauntOverAggro),
+				new_hate
 			);
-		}
-
-		if (success) {
-			if (hate_top && hate_top != this) {
-				int64 new_hate = (
-					(who->GetNPCHate(hate_top) - who->GetNPCHate(this)) +
-					bonus_hate +
-					RuleI(Combat, TauntOverAggro) +
-					1
-				);
-
-				LogHate(
-					"Not Top Hate - Taunter [{}] Target [{}] Hated Top [{}] Hate Top Amt [{}] This Character Amt [{}] Bonus_Hate Amt [{}] TauntOverAggro Amt [{}] - Total [{}]",
-					GetMobDescription(),
-					who->GetMobDescription(),
-					hate_top->GetMobDescription(),
-					who->GetNPCHate(hate_top),
-					who->GetNPCHate(this),
-					bonus_hate,
-					RuleI(Combat, TauntOverAggro),
-					new_hate
-				);
-
-				who->CastToNPC()->AddToHateList(this, new_hate);
-				success = true;
-			} else {
-				who->CastToNPC()->AddToHateList(this, 12);
-			}
 
-			if (who->CanTalk()) {
-				who->SayString(SUCCESSFUL_TAUNT, GetCleanName());
-			}
+			who->CastToNPC()->AddToHateList(this, new_hate);
 		} else {
-			MessageString(Chat::SpellFailure, FAILED_TAUNT);
+			LogHate("Already Hate Top");
+			who->CastToNPC()->AddToHateList(this, 12);
+		}
+
+		if (who->CanTalk()) {
+			who->SayString(SUCCESSFUL_TAUNT, GetCleanName());
 		}
+
+		MessageString(Chat::Skills, TAUNT_SUCCESS, who->GetCleanName());
 	} else {
-		MessageString(Chat::SpellFailure, FAILED_TAUNT);
+		MessageString(Chat::Skills, FAILED_TAUNT);
 	}
 
-	TryCastOnSkillUse(who, EQ::skills::SkillTaunt);
+	// Modern Abilities
+	if (RuleB(Combat, ClassicTauntSystem)) {
```

Comment:
If this section is supposed to be for modern abilities, wouldn't this be `if (!RuleB(Combat, ClassicTauntSystem)) {` ?

Reply:
Good Catch, thank you.

Reply:
Wonder if I should rule this off separately or remove the rule all together. This is less for the taunting itself and more for additional features.

Reply:
Generally - the more you can avoid deeply nested if/else chains the better. It quickly makes code harder to read and reason about. For this change specifically I would suggest the following

```cpp
	// classic taunt did not utilize modern abilities such as skill procs
	if (RuleB(Combat, ClassicTauntSystem)) {
		return;
	}
```

Reply:
![image](https://github.com/EQEmu/Server/assets/3319450/b601ad78-8b5b-4be1-8569-75e1f3736db7)

Reply:
I agree with cleaning up nested ifs for sure.

What I am referencing is the `TrySkillProc` is less directly tied to taunt or classic taunt and is a separate system and I am thinking that it shouldn't be ruled out as part of the classic taunt system. If someone wants to have SkillProcs from Taunt and still use the classic taunt system they should be able to.